src/check.h.in

```diff
@@ -669,6 +671,396 @@ CK_DLL_EXP void CK_EXPORT _ck_assert_failed(const char *file, int line,
+ * Check two floating point numbers to determine if X>≈Y with specified
+ * tolerance level
+ *
+ * If not X >≈ Y with error < powl(10, P), the test fails.
+ *
+ * @param X floating point number
+ * @param Y floating point number to compare against X
+ * @param P power of 10 (integer) as tolerance, if negative number is used
+ * it would mean number of decimal places (precision)
+ *
+ * @note If the check fails, the remaining of the test is aborted
+ *
+ * @since 0.11.0
+ */
+#define ck_assert_floating_ge_tol10(X, Y, P) _ck_assert_floating_op_tol10(X, >, Y, P, -1)
+
+/**
+ * Check two floating point numbers to determine if X<≈Y with specified
+ * tolerance level
+ *
+ * If not X <≈ Y with error < powl(10, P), the test fails.
+ *
+ * @param X floating point number
+ * @param Y floating point number to compare against X
+ * @param P power of 10 (integer) as tolerance, if negative number is used
+ * it would mean number of decimal places (precision)
+ *
+ * @note If the check fails, the remaining of the test is aborted
+ *
+ * @since 0.11.0
+ */
+#define ck_assert_floating_le_tol10(X, Y, P) _ck_assert_floating_op_tol10(X, <, Y, P, 1)
+
+/**
+ * Check floating point number to be finite (none of those: +infinity,
+ * -infinity and NaN)
+ *
+ * If X is +INFINITY or X is -INFINITY, or X is NaN, the test fails.
+ *
+ * @param X floating point number to be checked
+ *
+ * @note If the check fails, the remaining of the test is aborted
+ *
+ * @since 0.11.0
+ */
+#define ck_assert_floating_finite(X) \
+do { \
+  long double _ck_x = (X); \
+  ck_assert_msg(isfinite(_ck_x), \
+    "Assertion '%s' failed: %s == %.*Lg", \
+    #X" is finite", \
+    #X, (int)CK_FLOATING_DIG, _ck_x); \
+} while (0)
+
+/**
+ * Check floating point number to be infinite (+infinity or -infinity)
+ *
+ * If X is not +INFINITY and X is not -INFINITY, the test fails.
+ *
+ * @param X floating point number to be checked
+ *
+ * @note If the check fails, the remaining of the test is aborted
+ *
+ * @since 0.11.0
+ */
+#define ck_assert_floating_infinite(X) \
+do { \
+  long double _ck_x = (X); \
+  ck_assert_msg(isinf(_ck_x), \
+    "Assertion '%s' failed: %s == %.*Lg", \
+    #X" is infinite", \
+    #X, (int)CK_FLOATING_DIG, _ck_x); \
+} while (0)
+
+/**
+ * Check floating point number to be NaN
+ *
+ * If X is not NaN, the test fails.
+ *
+ * @param X floating point number to be checked
+ *
+ * @note If the check fails, the remaining of the test is aborted
+ *
+ * @since 0.11.0
+ */
+#define ck_assert_floating_nan(X) \
+do { \
+  long double _ck_x = (X); \
+  ck_assert_msg(isnan(_ck_x), \
+    "Assertion '%s' failed: %s == %.*Lg", \
+    #X" is NaN", \
+    #X, (int)CK_FLOATING_DIG, _ck_x); \
+} while (0)
+
+/**
+ * Check floating point number not to be NaN
+ *
+ * If X is NaN, the test fails.
+ *
+ * @param X floating point number to be checked
+ *
+ * @note If the check fails, the remaining of the test is aborted
+ *
+ * @since 0.11.0
+ */
+#define ck_assert_floating_nonnan(X) \
```

Comment:
I've just made as others do writing nonnull. It a special value such as the null constant. Real is just another name of float type in another languages (like Pascal). So such naming could be confusing. May be something like _correct/_incorrect? Since NaN indicates error in calculations.